zerver/lib/message.py

```diff
@@ -1435,4 +1435,4 @@ def set_visibility_policy_possible(user_profile: UserProfile, message: Message)
 
 def remove_single_newlines(content: str) -> str:
     content = content.strip("\n")
-    return re.sub(r"(?<!\n)\n(?![\n0-9*-])", " ", content)
+    return re.sub(r"(?<!\n)\n(?!\n|[-*] |[0-9]\. )", " ", content)
```

Comment:
We should catch multi-digit numbered lists:
```suggestion
    return re.sub(r"(?<!\n)\n(?!\n|[-*] |[0-9]+\. )", " ", content)
```

Reply:
That's right. Thanks for catching that.

Reply:
The commit message still says "a digit" but that's a nitpick.